test.py

```diff
@@ -337,32 +337,28 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
 
     def topology_for_class(self, class_name: str, cfg: dict) -> Callable[[], Awaitable]:
 
-        def create_server(cluster_name, seed):
-            cmdline_options = self.cfg.get("extra_scylla_cmdline_options", [])
-            if type(cmdline_options) == str:
-                cmdline_options = [cmdline_options]
-            server = ScyllaServer(
-                exe=self.scylla_exe,
-                vardir=os.path.join(self.options.tmpdir, self.mode),
-                host_registry=self.hosts,
-                cluster_name=cluster_name,
-                seed=seed,
-                cmdline_options=cmdline_options)
-
-            # Suite artifacts are removed when
-            # the entire suite ends successfully.
-            self.artifacts.add_suite_artifact(self, server.stop_artifact)
-            if not self.options.save_log_on_success:
-                # If a test fails, we might want to keep the data dir.
-                self.artifacts.add_suite_artifact(self, server.uninstall_artifact)
-            self.artifacts.add_exit_artifact(self, server.stop_artifact)
-            return server
-
         if class_name.lower() == "simple":
             async def create_cluster():
-                cluster = ScyllaCluster(int(cfg["replication_factor"]),
-                                        create_server)
+                cmdline_options = self.cfg.get("extra_scylla_cmdline_options", [])
+                if isinstance(cmdline_options, str):
+                    cmdline_options = [cmdline_options]
+                config_options = self.cfg.get("extra_scylla_config_options",
+                                              {"authenticator": "AllowAllAuthenticator",
+                                               "authorizer": "AllowAllAuthorizer"})
+                cluster = ScyllaCluster(self.scylla_exe,
+                                        int(cfg["replication_factor"]),
+                                        os.path.join(self.options.tmpdir, self.mode),
+                                        cmdline_options,
+                                        config_options,
+                                        self.hosts,
+                                        self.artifacts)
                 await cluster.install_and_start()
```

Comment:
Previously, cluster artifacts were added before install_and_start(), which can easily fail. Now they are added after. I think we should not leak clusters in any cases. Unless there is a strong reason to do otherwise, I suggest artifacts are added before the resources they protect are allocated.

Reply:
Reverted to previous Cluster and Server logic.